### DISCUSSION TITLE
Scone input files build and run in ubuntu 20.04 ?

### DISCUSSION THREAD

**Discussion by emilmammadzada99** [CambridgeNuclear/SCONE#25] [User Support]
How can i build and run in scone input file sample?Which with commandlines?
emil@DESKTOP-:~/SCONE/InputFiles$ l
BenchmarksMCNP/  JEZ  POPSY  SCONE_Inf  SCONE_Slab  XS/  c5g7_mox  mox_vol  sphere_with_DT  volCalc
emil@DESKTOPU-:~/SCONE/InputFiles$ ?

**Comment by Mikolaj-A-Kowalski** [comment]
Hi @emillon779! Thanks for the question! 
We try to keep the `Issue` tab to things related more to the development of the code rather than instructions on how to use it. I will transfer the thread to the `Discussion` tab. I hope you don't mind :-)

**Reply by emilmammadzada99** [reply]
How do I need to use command line to run the input file. The instructions are not enough

**Reply by Mikolaj-A-Kowalski** [reply]
So generally I assume you have successfully cloned the repository and have all the dependencies installed (LAPACK, pfUnit etc.). We tried to make some instructions on how one can download and compile all the dependencies (https://scone.readthedocs.io/en/latest/Installation.html#requirements), I don't know if you were able to find them? If there are any mistakes or problems please let us know! 

Once you have the environment compiling scone should just be the question of the following commands from the root of the repository 
```
mkdir build 
cd build 
cmake ... 
make -j 
```    

In second step you need to get nuclear data in ACE format (e.g. https://www.oecd-nea.org/dbdata/jeff/jeff33/index.html). 
When you unpack the archive you need to create a library file. It can be done with a command like that: 
```
scone_repo/scripts/make_ace_lib.sh ~/SCONE_data.aceXS CE /path-to-ace-files/*.ace 
```
Finally in the example input file (e.g. JEZ) you need to change the `aceLibrary /home/mak60/myACE/JEF311.aceXS;` entry to a file generated by the script in the previous step e.g. `aceLibrary /home/myusername/SCONE_data.aceXS;`. 

Afterwards SCONE should run with the following command  
``` 
scone_repo/build/scone.out scone_repo/InputFiles/JEZ
``` 

I hope it will work and let us know if you run into any problems :wink:

**Reply by emilmammadzada99** [reply]
App gave me this error ((
emil@DESKTOP-:~/SCONE/build$ cmake ..
-- /usr/lib/x86_64-linux-gnu/libopenblas.so-lm-ldl
CMake Error at /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  pFUnit unit test framework was not found.  Set PFUNIT_INSTALL environmental


  variable to the root of the pFUnit installation directory.  If pFUnit is
  not installed on your system, in README you will find instruction on how to
  obtain it and compile it (missing: PFUNIT_LIBRARY PFUNIT_MODULES
  PFUNIT_PREPROC)
Call Stack (most recent call first):
  /usr/share/cmake-3.22/Modules/FindPackageHandleStandardArgs.cmake:594 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindPFUNIT.cmake:78 (find_package_handle_standard_args)
  CMakeLists.txt:86 (find_package)

**Reply by Mikolaj-A-Kowalski** [reply]
This happens when cmake cannot find pFUnit. It is only required to run the unit and integration tests so the easiest way to avoid this error is to compile without tests. It can be done with: 
```
mkdir build
cd build
cmake .. -DBUILD_TESTS=OFF
make -j 
```

**Reply by emilmammadzada99** [reply]
i m using pfunit v3.3  app running.Do we need to transfer ace data to SCONE_data.aceXS folder one-by-one?

**Reply by Mikolaj-A-Kowalski** [reply]
No. It is just convenient to have all ace files in a single folder but it is not necessary. 
The `SCONE_data.aceXS` file from my example earlier should not be a folder but a file when each line lists the name of an ACE card, line number when it begins and a path to the file where it is contained. It should be produced by the  `make_ace_lib.sh` script which uses `awk` to search files for ACE cards in the files. 
The command:
```
scone_repo/scripts/make_ace_lib.sh ~/SCONE_data.aceXS CE /path-to-ace-files/*.ace 
```    
Takes the name of the library file as a 1st argument. 
Type of the data (CE or SAB for thermal scattering libraries) as a 2nd argument
The end is a list of all files to search for data. It can be multiple arguments. It is just more convenient to specify multiple cards with the UNIX `*` wildcard.

**Reply by emilmammadzada99** [reply]
Understood, thank you))) Also thank you for developing such a project

**Reply by Mikolaj-A-Kowalski** [reply]
Thanks for using it! ;-) 
Again if you run into any problems or issues do not hesitate to let us know.

**Reply by emilmammadzada99** [reply]
How can I access the output files after running the program?
`<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
                                                                         Fatal Error has occured in:
                                                                           load (aceLibrary_mod.f90)

                                                                           For the following reason:
File Error: Cannot open file '/home/mak60/myACE/JEF311.aceXS': No such file or directory

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>`

**Reply by Mikolaj-A-Kowalski** [reply]
Actually it crashed because it was looking for a ACE library file from the script at "'/home/mak60/myACE/JEF311.aceXS" you need to change in the input file to path on your system. 

JEZ should produce `output.m` file after successful run, which can be read with MATLAB. Alternatively you can use JSON format that can be used in tandem with Python, see: https://scone.readthedocs.io/en/latest/Output.html for details
To change format to JSON you need to add `outputFormat asciiJSON;` entry to the input file.

**Reply by emilmammadzada99** [reply]
what this error?
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
                                                                         Fatal Error has occured in:
                                                                 new_neutronACE (aceLibrary_mod.f90)

                                                                           For the following reason:
31000.03 was not found in ACE library from: /home/emil/SCONE_data.aceXS

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>

**Comment by Mikolaj-A-Kowalski** [comment]
Cursed Gallium! 
Basically the JEZ was beeing run on our side with JEFF 3.1 data which had this special nuclide 31000.03 which represents a 'natural' isotopic mix of Gallium. Clearly it has been replaced with individual isotopes in newer JEFF 3.3. 
Changing `31000.03` inside the fuel composition in input file to the most common gallium isotope `31069.03` should do the trick 

Basically the input file looks for the `31000.03c` ACE card in the library file, but it is not there. This is why SCONE complains.

**Reply by emilmammadzada99** [reply]
From which table can I find the equivalent of this isotope in jeff3.3 to change it in the code?

**Reply by Mikolaj-A-Kowalski** [reply]
The isotopes in ACE are specified using ZZAAA.TT convention where ZZ represents atomic number, AAA mass number and TT a code to distinguish different versions of a card (usually TT represents approximate temperature of the target material in 10s of K) 

Thus 31069.03 would usually be Gallium-69 at 273K. 
As for the isotopic composition of Gallium Wikipedia can serve as a quick reference (https://en.wikipedia.org/wiki/Gallium). 
For practical purposes IUPAC report can serve as a source (https://iupac.org/body/210/) (https://www.degruyter.com/document/doi/10.1515/pac-2015-0503/html)

**Reply by emilmammadzada99** [reply]
I encountered this error when I ran the scone_slab instance. How do I solve this problem?
`emil@DESKTOP-:~/SCONE$ ./Build/scone.out ./InputFiles/SCONE_Slab
  ><((((*>  ><((((*>  ><((((*>  ><((((*>  ><((((*>  ><((((*>  ><((((*>  ><((((*>  ><((((*>  ><((((*>

         _____ __________  _   ________
        / ___// ____/ __ \/ | / / ____/
        \__ \/ /   / / / /  |/ / __/
       ___/ / /___/ /_/ / /|  / /___
      /____/\____/\____/_/ |_/_____/


 Compiler Info :   GCC version 9.4.0
 OpenMP Threads:    1

  <*((((><  <*((((><  <*((((><  <*((((><  <*((((><  <*((((><  <*((((><  <*((((><  <*((((><  <*((((><
 <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
 /\/\ READING GEOMETRY /\/\
 Building Surfaces
 DONE!
 Building Cells
 DONE!
 Building Universes
 DONE!
 CHECKING GEOMETRY:
  Recursion in definition - NOT PRESENT!
  Nesting level - FINE!
  Outside below root - NOT PRESENT!
 BUILDING GEOMETRY GRAPH
 DONE!
 GEOMETRY INFORMATION
  Number of Surfaces: 1
  Number of Cells: 0
  Number of Universes: 1
  Nesting Levels: 1
  Unique Cells: 1
  Unused universes (ID):
  Boundary Surface ID: 1
  Boundary Surface Type: box
  Boundary Conditions: 0 0 1 1 1 1
 \/\/ FINISHED READING GEOMETRY \/\/
 <><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
Building material: fuel From: ./XS/URRa_2_1_XSS
<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>
                                                                         Fatal Error has occured in:
                                                                    fileToDict (dictParser_func.f90)

                                                                           For the following reason:
                                     Cannot open file './XS/URRa_2_1_XSS': No such file or directory

<><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><><>`

**Reply by ChasingNeutrons** [reply]
This error is because the input file is expecting the cross section input to be in the folder where you are running things which in your case is the SCONE folder. The cross section input sits inside the InputFiles folder. If you run scone from in there, i.e.,:
`cd ./InputFiles/`
`../Build/scone.out SCONE_Slab`
then it should run fine.

**Reply by emilmammadzada99** [reply]
yes it worked right now the application, thank you. Should I run all the sample files with this command?
